### DISCUSSION TITLE
Help using protobuf-uml-diagram

### DISCUSSION THREAD

**Discussion by Hdd87** [kinow/protobuf-uml-diagram#62] [General]
Hi!
Thank you for replying to my message on insta!

My goal is to create UML class diagrams from proto files in the following repository: 
 https://github.com/th2-net ,

 an example proto file is : 
https://github.com/th2-net/th2-grpc-common/blob/master/src/main/proto/th2_grpc_common/common.proto

My machine is using windows 10 .  

**what I did so far:**

1. I cloned all the repositories I needed into my local machine and then I separated the .proto files into a separate folder . 
2. Next using CMD I ran  `pip install protobuf-uml-diagram`   and it looks like the installation was okay since the message says "Requirement already satisfied" .( attached a snap shot below)

![image](https://user-images.githubusercontent.com/102023206/220313331-708231b5-61e9-4de4-9ddd-e7b3671ab740.png)

3. Next I ran protobuf-uml-diagram and I received this :

![image](https://user-images.githubusercontent.com/102023206/220313731-510e1061-2b9e-4f56-ba5a-4b95826a7a4e.png)

**Questions :**

1. what are the next steps to take to proceed with creating some UML class diagrams 
2. what are the sections USAGE , Logging output, and TL;DR used for ? 
3. do you have any videos about using your repository? :)

Thank you !

**Comment by kinow** [comment]
Deleted #63 as it was duplicating this one.

**Reply by kinow** [reply]
Hi @Hdd87 

>pip install protobuf-uml-diagram

The command above should have installed the module and put the `protobuf-uml-diagram` utility in your `$PATH` (or `%PATH%` on Windows).

I guess it could be something related to Windows perhaps, or maybe the tool was not installed successfully?

Do you see the module when you run `pip list`?

**Reply by Hdd87** [reply]
Hi @kinow 

yes I can see the module 

![image](https://user-images.githubusercontent.com/102023206/220345951-544a0f5b-82fb-436f-855c-8d9e8609f27b.png)

**Reply by kinow** [reply]
Thanks.

Does this command exit successfully? `python -c 'import protobuf_uml_diagram'`

It will import the protobuf_uml_diagram`. If this command works, but `protobuf-uml-diagram` doesn't, that will confirm the module is installed but the script is not.

**Reply by Hdd87** [reply]
@kinow  I typed this command in CMD and this was the result 

![image](https://user-images.githubusercontent.com/102023206/220360965-05a89407-5a3c-4b12-be8f-0bc32b454bb7.png)

**Reply by kinow** [reply]
`protobuf_uml_diagram` is the module. The program is `protobuf-uml-diagram`.

If you were using a virtual environment I think it would be easier to locate the directory with the script, but on Linux it would be something like `~/.local/bin`. Maybe you can check where these scripts get installed on Windows and verify if the `protobuf-uml-diagram` exists where expected.

**Reply by Hdd87** [reply]
Thanks @kinow !  I did pip show  command and 

![image](https://user-images.githubusercontent.com/102023206/220364629-9dede59f-ffac-4ea8-a80f-e14ecb783763.png)

if I was using a linux machine  what would we be the proper steps to follow to get the program to start running ?
